source/MaterialXContrib/MaterialXNode/MaterialXShadingNodeImpl.cpp

```diff
@@ -232,99 +232,53 @@ void MaterialXShadingNodeImpl<BASE>::updateShader(MHWRender::MShaderInstance& sh
 
     MaterialX::DocumentPtr document = node->materialXData->getDocument();
     const MaterialX::StringMap& inputs = node->materialXData->getPathInputMap();
-    for (auto i : inputs)
+    for (const auto& input : inputs)
```

Comment:
Yet another instance of the auto antipattern. I fixed it once and it was stomped by a merge. Also, "i" looked too much like an index.